File: app/src/main/java/com/namae0Two/khmeralternativekeyboard/KhmerKeyboardService.java
package com.namae0Two.khmeralternativekeyboard;

import android.inputmethodservice.InputMethodService;
import android.view.View;
import android.view.inputmethod.InputConnection;
import android.widget.LinearLayout;

import com.namae0Two.khmeralternativekeyboard.view.KeyboardView;

public class KhmerKeyboardService extends InputMethodService {

    private KeyboardView mKeyboardLayout;



    @Override
    public View onCreateInputView() {

        mKeyboardLayout = new KeyboardView(getApplicationContext(), this);


        return mKeyboardLayout;
    }

    @Override
    public void onComputeInsets(Insets outInsets) {
        super.onComputeInsets(outInsets);

        //this set to enable popup window top of the keyboardl ayout by pass touch through the top
        //placeholder view
        outInsets.visibleTopInsets = mKeyboardLayout.getKeyboardContent().getTop();
        outInsets.contentTopInsets = mKeyboardLayout.getKeyboardContent().getTop();
    }
}

File: app/src/main/java/com/namae0Two/khmeralternativekeyboard/KhmerKeyboardService.kt
package com.namae0Two.khmeralternativekeyboard

import android.inputmethodservice.InputMethodService
import android.view.View
import com.namae0Two.khmeralternativekeyboard.database.DictionaryWordDatabase
import com.namae0Two.khmeralternativekeyboard.view.KeyboardView

class KhmerKeyboardService : InputMethodService() {

    private var mKeyboardLayout: KeyboardView? = null
    private var database: DictionaryWordDatabase? = null

    override fun onCreateInputView(): View {

        if (database == null) {
            database = DictionaryWordDatabase.getInstance(application)
        }

        mKeyboardLayout = KeyboardView(applicationContext, this)



        return mKeyboardLayout!!
    }

    override fun onComputeInsets(outInsets: Insets) {
        super.onComputeInsets(outInsets)

        //this set to enable popup window top of the keyboardl ayout by pass touch through the top
        //placeholder view
        outInsets.visibleTopInsets = mKeyboardLayout!!.keyboardContent.top
        outInsets.contentTopInsets = mKeyboardLayout!!.keyboardContent.top
    }

}